types/node/stream.d.ts

```diff
@@ -22,25 +22,26 @@ declare module "node:stream" {
     import { Abortable, EventEmitter } from "node:events";
     import * as promises from "node:stream/promises";
     import * as web from "node:stream/web";
-
-    type ComposeFnParam = (source: any) => void;
-
     class Stream extends EventEmitter {
+        /**
+         * @since v0.9.4
+         */
         pipe<T extends NodeJS.WritableStream>(
             destination: T,
-            options?: {
-                end?: boolean | undefined;
-            },
-        ): T;
-        compose<T extends NodeJS.ReadableStream>(
-            stream: T | ComposeFnParam | Iterable<T> | AsyncIterable<T>,
-            options?: { signal: AbortSignal },
+            options?: Stream.PipeOptions,
         ): T;
```

Comment:
This is definitely not a method on the legacy Stream prototype, and the definition was extremely dodgy. Moved to the Readable prototype and expanded to use the new helper types.